storage/innobase/btr/btr0cur.cc

```diff
@@ -973,26 +973,41 @@ static int btr_latch_prev(buf_block_t *block, page_id_t page_id,
  retry:
   /* Pass no_wait pointer to ensure that we don't wait on the current page
   latch while holding the next page latch to avoid latch ordering violation. */
-  bool no_wait= false;
+  bool would_block= false;
```

Comment:
I thought that the comment refers to the name of the function parameter `bool *no_wait`, because it mentions "pointer".

I just realized that the parameter `allow_ibuf_merge=false` is incorrect. We are responsible for merging any changes to the page. I think that I will refactor this to use the `buf_pool_t::page_fix()` function that was introduced in b7b9f3ce825a08f16ad851c16f603365ae473195 (#3448). In that case, we’d also have to relocate a compressed-only `ROW_FORMAT=COMPRESSED` page in case it is one.

Reply:
418147edc388f82e452c9bc93d686de9027ad219 simplifies the interface by extending `buf_pool_t::page_fix()` to cover `ROW_FORMAT=COMPRESSED` pages as well as the change buffer. There is a parameter `bool no_wait` and a special return value `-1` that indicates that the call would block.

Reply:
Sorry, e1fd750d1522c64d1265f9dd84b6c046afea4c25 fixes a logic error that 418147edc388f82e452c9bc93d686de9027ad219 left in `btr_latch_prev()`.